src/coreclr/inc/corinfoinstructionset.h

```diff
@@ -128,41 +128,77 @@ enum CORINFO_InstructionSet
 struct CORINFO_InstructionSetFlags
 {
 private:
-    uint64_t _flags = 0;
+    static const int FlagsArrSize = 1;
```

Comment:
If you defined a `BitsPerFlagsField` then you could have a helper for getting a given "field index" and then "relative bit" within that field.

For example:
```
static const int32_t FlagsFieldCount = 2;
    static const int32_t BitsPerFlagsField = sizeof(uint64_t) * 8;

    uint64_t _flags[FlagsFieldCount] = { };

    static int32_t GetFlagsFieldIndex(CORINFO_InstructionSet instructionSet)
    {
        int32_t bitIndex = (int32_t)instructionSet;
        return bitIndex / BitsPerFlagsField;
    }

    static uint64_t GetRelativeBitIndex(CORINFO_InstructionSet instructionSet)
    {
        return ((uint64_t)1) << (instructionSet & 0x3F);
    }

    void AddInstructionSet(CORINFO_InstructionSet instructionSet)
    {
        int32_t index = GetFlagsFieldIndex(instructionSet);
        _flags[index] |= GetRelativeBitIndex(instructionSet);
    }
```

This would allow less overall churn in the API surface over time while still being optimized down correctly by the native compiler.

Reply:
I really like this. Have made the change